Machine_Learning/README.md

```diff
@@ -624,7 +634,31 @@ All of the things above are "unwanted" but it is important to understand that me
+
+
+## Understanding the training parameters
+
+The Intellesis Denoising training interface shares a resemblance with the Intellesis machine learning training interface, making it user-friendly for both new and existing users to develop personalized denoising models. This training interface features four adjustable parameters, namely the number of epochs, batch size, window size, and masking ratio. While the default values may suffice, users may need to lower the batch size when using a resource-limited system (e.g., your work laptop). Further details on each of these parameters are available below.
+
+### Number of epochs:
+
+The "number of epochs" refers to the number of times the training dataset is passed through the neural network during the training process. Each pass through the dataset is called an epoch. In N2V, the number of epochs is a user-defined parameter that determines how many times the neural network will be trained on the full dataset. If the number of epochs is too short, the model may not be able to learn the underlying patterns in the data, resulting in underfitting. On the other hand, if the number of epochs is too long, the model may start to memorize the training data, resulting in overfitting making it less effective at denoising new images. 
+
+
+### Batch size:
+
+The "batch size" refers to the number of samples (image patches) that are processed by the neural network at once. During each epoch, the training dataset is typically divided into several smaller batches, and the neural network is updated based on the average loss across the batch. The batch size is another user-defined parameter that controls how many samples are processed in each batch. If the batch size is too small, the model may converge slowly and can result in poor generalization. If the batch size is too large, it may result in the model being unable to fit into memory, slowing down the training process, and potentially leading to poor generalization. The default batch size in ZEN is 64, but it is advisable to experiment with smaller sizes, such as 32 or 16, if memory errors occur.
+
+### Window size:
+
+The "window size" parameter determines the size of the image patch that N2V uses to make predictions. Specifically, N2V divides the input image into overlapping patches of a fixed size (determined by the window size parameter) and trains a neural network to predict the central pixel of each patch based on the other pixels in the patch. During inference (denoising), N2V processes the input image in a sliding-window fashion, applying the trained neural network to each patch in turn to produce a denoised output image. If the window size is too small, it may not capture enough information to properly denoise the image. If the window size is too large, it may include too much information, leading to overfitting, increased computation time, and reduced performance. The default window size of 5 is optimal for most images. 
```

Comment:
The window size does not have an effect on the patch size. Given a pixel to be masked, it determines the size of the surrounding area to sample replacement pixels from.

Reply:
I thought this is what what meant by Sreeni with his text, but you are right. I modified that phrase.